tests/baselines/reference/objectTypeWithStringAndNumberIndexSignatureToAny.errors.txt

```diff
@@ -0,0 +1,145 @@
+
+
+==== tests/cases/conformance/types/members/objectTypeWithStringAndNumberIndexSignatureToAny.ts (13 errors) ====
+    // When checking compatibility between two types,
+    // TypeScript should not require an index signature if
+    // the target side index signature maps to `any` *and*
+    // the target side has *any* string index signature to `any`.
+    //
+    // So an index signature like in
+    //
+    //  { [x: number]: any }
+    //
+    // is still required of a source type, but neither index signature in
+    //
+    //  { [x: number]: any, [x: string]: any; }
+    //
+    // should be required; *however*, the number index signature in
+    //
+    //  { [x: number]: number, [x: string]: any; }
+    //
+    // should always be required.
+    
+    interface StringTo<T> {
+        [x: string]: T;
+    }
+    
+    interface NumberTo<T> {
+        [x: number]: T;
+    }
+    
+    interface StringAndNumberTo<T> extends StringTo<T>, NumberTo<T> {
+    }
+    
+    interface Obj {
+        hello: string;
+        world: number;
+    }
+    
+    function f1(sToAny: StringTo<any>, nToAny: NumberTo<any>, bothToAny: StringAndNumberTo<any>, someObj: Obj) {
+        sToAny = nToAny;
+        sToAny = bothToAny;
+        sToAny = someObj;
+    
+        nToAny = sToAny;
+        nToAny = bothToAny;
+        nToAny = someObj;
+        ~~~~~~
+!!! error TS2322: Type 'Obj' is not assignable to type 'NumberTo<any>'.
+!!! error TS2322:   Index signature is missing in type 'Obj'.
+    
+        bothToAny = sToAny;
+        bothToAny = nToAny;
+        bothToAny = someObj;
+    
+        someObj = sToAny;
+        ~~~~~~~
+!!! error TS2739: Type 'StringTo<any>' is missing the following properties from type 'Obj': hello, world
+        someObj = nToAny;
+        ~~~~~~~
+!!! error TS2739: Type 'NumberTo<any>' is missing the following properties from type 'Obj': hello, world
+        someObj = bothToAny;
+        ~~~~~~~
+!!! error TS2739: Type 'StringAndNumberTo<any>' is missing the following properties from type 'Obj': hello, world
+    }
+    
+    function f2(sToAny: StringTo<any>, nToAny: NumberTo<any>, bothToAny: StringTo<any> & NumberTo<any>, someObj: Obj) {
+        sToAny = nToAny;
+        sToAny = bothToAny;
+        sToAny = someObj;
+    
+        nToAny = sToAny;
+        nToAny = bothToAny;
+        nToAny = someObj;
+        ~~~~~~
+!!! error TS2322: Type 'Obj' is not assignable to type 'NumberTo<any>'.
+    
+        bothToAny = sToAny;
+        bothToAny = nToAny;
+        bothToAny = someObj;
+        ~~~~~~~~~
+!!! error TS2322: Type 'Obj' is not assignable to type 'StringTo<any> & NumberTo<any>'.
+!!! error TS2322:   Type 'Obj' is not assignable to type 'NumberTo<any>'.
+!!! error TS2322:     Index signature is missing in type 'Obj'.

```

Comment:
This might be questionable. This used to work in 4.2, but this PR doesn't un-break this code.